src/main/scala/com/thirdship/libunit/AStarSolver.scala

```diff
@@ -101,29 +103,29 @@ case class AStarSolver(var allTSUnits: List[String], var allConversions: List[Co
       * @param  end      The ending unit form the search
       * @return A list of conversion edges that when listed in order describe the path of conversion from start to end
 	  */
-  private def reconstructConversionEdgePathList(cameFrom: Map[ String, String], end: String): List[ConversionEdge[String, Double, Double]] = {
+  private def reconstructConversionEdgePathList(cameFrom: mutable.Map[String, String], end: String): List[ConversionEdge[String, Double, Double]] = {
```

Comment:
What is the justification on using a mutable map?

I would prefer avoid mutation whenever is possible.